students/chieu_quach/lesson04/mailroom_part2.py

```diff
@@ -0,0 +1,133 @@
+                     
+                amount = float(value['Amount'])                
+                last_name = value['last name']
+                # added "_" to last name
+                last_name = ("_".join(last_name.split()))
+                fullname = value['first name'] + "_" + last_name
+                fullname = fullname + ".txt"
+                #print ("fullname ", fullname)       
+               
+            
+            #print("msg".format(name_list[0], name_list[1]))
+                print(msg.format(name, amount))
+            # sends text message to output file
+                outfile = open(fullname, "w")
+                cpy = (msg.format(name, amount))
+                outfile.writelines(cpy)
+                 
+def send_letters():
+
+     # reading key and value in list
+     # prints list of letters from donation list
+      for key, value in donation_list.items():            
+    
+                   
+          amount = float(value['Amount'])                
+          name   = value['first name'] + " " + value['last name']
+          amount = float(value['Amount'])                
+          last_name = value['last name']
+          # added "_" to last name
+          last_name = ("_".join(last_name.split()))
+          fullname = value['first name'] + "_" + last_name
+          fullname = fullname + ".txt"
+         # print ("fullname ", fullname)       
+    
+          print(msg.format(name, amount))
+          outfile = open(fullname, "w")
+          cpy = (msg.format(name, amount))
+          outfile.writelines(cpy)
+         # print ("cpy ", cpy)
+       
+          outfile.close()
+            
+      
+                                 
+            
+def create_report():
+
+   name_header  = " Donor Name "
+   total_given = " |  Total  Given   |"
+   num_gifts    = " Num Gifts      | "
+   average_gift = " Average  Gift  " 
+   print ("{:15}  {:15}  {:15}  {:15}"
+   .format(name_header, total_given, num_gifts, average_gift))
+   print (" ------------------------------------------------------------------------")
+   # reading key and value in list
+   for key, value in donation_list.items():            
+    
+         full_name = value['first name'] + " " + value['last name']    
+         list_name = (full_name, float(value['Amount']), int(value['num_gift']), float(value['avg_amt']))
+         print("{:<19} {:14.2f} {:15d}{:=21.2f} ".format(*list_name))
+         
+   print ("\n\n\n")
+    
+   
+def exit_program():
+   print ("Thanks for visiting ")
+   sys.exit()
+
+def main():
+
+       # Use switch case to call functions from user input
+       switch = {1: send_thankyou,
+                 2: create_report,
+                 3: send_letters,               
+                 4: exit_program,
+                          
+       }
+       response = input(prompt)
+       response = int(response)
+       switch[response]()
+       
+     
+
+# Main function
+if __name__ == "__main__": 
+  
+   main()
```

Comment:
@Chieu1 - this version isn't working as expected.  When sending a thank you to a single donor, if I enter a name, it breaks out of the program.  Also, the program should continue to loop until the user exits.  Also, there are a lot of PEP8 issues - mainly spacing.  All indentations should be 4 spaces in Python.